src/NetTopologySuite.IO.SqlServerBytes/SqlServerBytesReader.cs

```diff
@@ -238,6 +238,12 @@ private Geometry ToGeometry(Geography geography)
                         geometries.Remove(shapeIndex);
                         break;
 
+                    case OpenGisType.CircularString:
+                    case OpenGisType.CompoundCurve:
+                    case OpenGisType.CurvePolygon:
+                    case OpenGisType.FullGlobe:
+                        return new UnsupportedGeometry(shape.Type, geography);
```

Comment:
Note, if a GeometryCollection contains one of these, the whole collection will be returned as unsupported. A slightly better solution might be to return a GeometryCollection containing an UnsupportedGeometry.